src/Crocodealer/Core/Label.hs

```diff
@@ -0,0 +1,24 @@
+module Crocodealer.Core.Label
+       ( LabelRule(..)
+       , Label(..)
+       , Name(..)
+       , HexColor(..)
+       ) where
+
+newtype Name a = Name
+    { unName :: Text
+    }
+
+newtype HexColor = HexColor
+    { unHexColor :: Text
+    }
+
+data Label = Label
+    { labelName        :: Name Label
+    , labelDescription :: !Text
+    , labelColor       :: HexColor
+    }
+
+data LabelRule
+    = Create Label
+    | Override (Name Label) (Name Label)
```

Comment:
Would be great to add some documentation to this data type. This is the only non-trivial data type in this PR, so it's always good to add some docs to it.